cachito/workers/tasks/gitsubmodule.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+import tarfile
+import tempfile
+
+import git
+
+from cachito.errors import CachitoError
+from cachito.workers.pkg_managers.general import update_request_with_package
+from cachito.workers.tasks.celery import app
+from cachito.workers.paths import SourcesDir
+from cachito.workers.scm import Git
+
+__all__ = ["add_git_submodules_as_package"]
+log = logging.getLogger(__name__)
+
+
+@app.task
+def add_git_submodules_as_package(request_id, url, ref):
+    """
+    Add git submodules as package to the Cachtio request.
+
+    :param int request_id: the Cachito request ID this is for
+    :param str url: the source control URL to pull the source from
+    :param str ref: the source control reference
+    :raises CachitoError: if adding submodules as a package fail.
+    """
+    scm = Git(url, ref)
+    sources_dir = SourcesDir(scm.repo_name, f"{scm.ref}-with-submodules")
+    archive_path = sources_dir.archive_path
+    with tempfile.TemporaryDirectory(prefix="cachito-") as temp_dir:
+        with tarfile.open(archive_path, mode="r:gz") as tar:
+            tar.extractall(temp_dir)
+        repo = git.Repo(os.path.join(temp_dir, "app"))
+        for sm in repo.submodules:
+            try:
+                # Save package to db
+                package = {
+                    "type": "git-submodule",
+                    "name": sm.name,
+                    "version": f"{sm.url}#{sm.hexsha}",
+                }
+                log.debug("Adding submodule '%s' as a package for Cachito request", sm.name)
+                update_request_with_package(request_id, package)
+            except Exception as e:
```

Comment:
`update_request_with_package` raises a CachitoError when an API call to update the request fails. Can we capture specific exceptions here? 

If something else fails for some other unexpected reason, we should consider it to be a bug and tend to it. Letting it fail without capturing it here may help debugging (and should definitely call our attention to it), when/if it happens.